app/scripts/modules/core/src/pipeline/config/stages/entityTags/TagEditor.tsx

```diff
@@ -0,0 +1,169 @@
+import * as React from 'react';
+import { set } from 'lodash';
+import Select, { Option } from 'react-select';
+
+import { IEntityTag } from 'core/domain';
+import { HelpField } from 'core/help/HelpField';
+import { UUIDGenerator } from 'core/utils';
+
+import './TagEditor.less';
+
+export type EntityTagType = 'notice' | 'alert' | 'custom';
+
+export interface ITagEditorProps {
+  onChange: (tag: IEntityTag) => void;
+  tag: IEntityTag;
+}
+
+export interface ITagEditorState {
+  type: EntityTagType;
+  valueIsObject: boolean;
+}
+
+const typeOptions: Array<Option<EntityTagType>> = [
+  { label: 'Notice', value: 'notice' },
+  { label: 'Alert', value: 'alert' },
+  { label: 'Custom', value: 'custom' },
+];
+
+export class TagEditor extends React.Component<ITagEditorProps, ITagEditorState> {
+  constructor(props: ITagEditorProps) {
+    super(props);
+
+    const type = props.tag.name.startsWith('spinnaker_ui_notice')
+      ? 'notice'
+      : props.tag.name.startsWith('spinnaker_ui_alert')
+        ? 'alert'
+        : 'custom';
+
+    const valueIsObject = typeof props.tag.value === 'object';
+
+    this.state = {
+      type,
+      valueIsObject,
+    };
+  }
+
+  public componentWillReceiveProps(nextProps: ITagEditorProps) {
```

Comment:
suggest combining this with the code in the constructor and moving it all into `componentDidUpdate`

Reply:
👍